lib/posix/clock.c

```diff
@@ -195,3 +196,24 @@ int gettimeofday(struct timeval *tv, void *tz)
 
 	return res;
 }
+
+/**
+ * @brief Get current real time.
+ *
+ * See IEEE 1003.1
+ */
+int clock_getcpuclockid(pid_t pid, clockid_t *clock_id)
+{
+	/* We don't allow any process ID but our own.  */
+	if (pid != 0 && pid != getpid()) {
+		return EPERM;
+	}
+
+#ifdef CLOCK_PROCESS_CPUTIME_ID
+	*clock_id = CLOCK_PROCESS_CPUTIME_ID;
+	return 0;
+#endif
+
+	/* We don't have a timer for that.  */
+	return ENOENT;
```

Comment:
It's probably safe to simplify now that `CLOCK_PROCESS_CPUTIME_ID` is defined
```suggestion
	*clock_id = CLOCK_PROCESS_CPUTIME_ID;
	return 0;
```

Reply:
Addressed your comments.